cdap-app-templates/cdap-etl/hydrator-spark-core/src/main/java/co/cask/cdap/etl/spark/StreamingCompat.java

```diff
@@ -44,8 +47,11 @@ private StreamingCompat() {
 
   }
 
-  public static JavaStreamingContext getOrCreate(String checkpointDir, Function0<JavaStreamingContext> contextFunc) {
-    return JavaStreamingContext.getOrCreate(checkpointDir, contextFunc);
+  public static JavaStreamingContext getOrCreate(String checkpointDir, Function0<JavaStreamingContext> contextFunc,
+                                                 @Nullable Configuration configuration) {
+    return configuration == null
```

Comment:
couldn't tell from the code when this is null. Do we ever expect it to be null? Won't the defaultFs not be used in that case?

Reply:
yeah good point, it was null in a previous iteration of the code